compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -955,6 +960,94 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Recover from `Fn`-family traits (Fn, FnMut, FnOnce) with lifetime arguments
+    /// (e.g. `FnOnce<'a>(&'a str) -> bool`). Up to generic arguments have already
+    /// been eaten.
+    fn recover_fn_trait_with_lifetime_params(
+        &mut self,
+        fn_path: &mut ast::Path,
+        lifetime_defs: &mut Vec<GenericParam>,
+    ) -> PResult<'a, ()> {
+        let Some(fn_path_segment) = fn_path.segments.last_mut() else {
```

Comment:
You could just use `.unwrap()` here instead of `let`/`else`.

Reply:
Thanks, I've tried. But I think it'll be more difficult to get `inputs_span` when using `parse_fn_decl`:

```rust
// Here, inputs refers their own spans, but not include "(" and ")".
let ast::FnDecl { inputs, output } = self.parse_fn_decl(...)?.into_inner();
```

Any better way here?